### PR TITLE
log4cplus: add version 2.0.8, support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **log4cplus/***

- add version 2.0.8
- support conan v2
- update libiconv

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

File: recipes/log4cplus/all/conandata.yml [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
   "2.0.7":
-    url: "https://github.com/log4cplus/log4cplus/releases/download/REL_2_0_7/log4cplus-2.0.7.tar.bz2"
-    sha256: "8fadbafee2ba4e558a0f78842613c9fb239c775d83f23340d091084c0e1b12ab"        
+    - patch_file: "patches/2.0.6-0001-fix-cmake.patch"
+      patch_description: "disable fPIC, move cmake_minimum_version to front"
+      patch_type: "backport"
+  "2.0.6":
+    - patch_file: "patches/2.0.6-0001-fix-cmake.patch"
+      patch_description: "disable fPIC, move cmake_minimum_version to front"
+      patch_type: "backport"
```
Comment (SpaceIm) [inline recipes/log4cplus/all/conandata.yml:30]:
```suggestion
      patch_type: "conan"
```

Comment (SpaceIm) [reply recipes/log4cplus/all/conandata.yml:30]: same for all other patches

Comment (toge) [reply recipes/log4cplus/all/conandata.yml]: @SpaceIm 
Thank you.
Fixed.

File: recipes/log4cplus/all/conanfile.py [hunk 2]
```diff
@@ -1,110 +1,106 @@
...
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.libs = collect_libs(self)
```
Comment (SpaceIm) [inline recipes/log4cplus/all/conanfile.py:100]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "log4cplus")
        self.cpp_info.set_property("cmake_target_name", "log4cplus::log4cplus")
        self.cpp_info.libs = collect_libs(self)
```

There is config file upstream, so these properties can be enforced.

**Review by SpaceIm — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**